azurerm/internal/services/network/tests/resource_arm_virtual_hub_connection_test.go

```diff
@@ -0,0 +1,272 @@
+	template := testAccAzureRMVirtualHubConnection_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_virtual_hub_connection" "test" {
+  name                      = "acctestvhubconn-%d"
+  virtual_hub_id            = azurerm_virtual_hub.test.id
+  remote_virtual_network_id = azurerm_virtual_network.test.id
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMVirtualHubConnection_requiresImport(data acceptance.TestData) string {
+	template := testAccAzureRMVirtualHubConnection_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_virtual_hub_connection" "import" {
+  name                      = azurerm_virtual_hub_connection.test.name
+  virtual_hub_id            = azurerm_virtual_hub_connection.test.virtual_hub_id
+  remote_virtual_network_id = azurerm_virtual_hub_connection.test.remote_virtual_network_id
+}
+`, template)
+}
+
+func testAccAzureRMVirtualHubConnection_complete(data acceptance.TestData) string {
+	template := testAccAzureRMVirtualHubConnection_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_virtual_hub_connection" "test" {
+  name                                       = "acctestvhubconn-%d"
+  virtual_hub_id                             = azurerm_virtual_hub.test.id
+  remote_virtual_network_id                  = azurerm_virtual_network.test.id
+  allow_hub_to_remote_vnet_transit           = true
+  allow_remote_vnet_to_use_hub_vnet_gateways = false
+  enable_internet_security                   = true
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMVirtualHubConnection_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestrg-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctestvirtnet%d"
+  address_space       = ["10.5.0.0/16"]
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_network_security_group" "test" {
+  name                = "acctestnsg%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_subnet" "test" {
+  name                 = "acctestsubnet%d"
+  resource_group_name  = azurerm_resource_group.test.name
+  virtual_network_name = azurerm_virtual_network.test.name
+  address_prefix       = "10.5.1.0/24"
+}
+
+resource "azurerm_subnet_network_security_group_association" "test" {
+  subnet_id                 = azurerm_subnet.test.id
+  network_security_group_id = azurerm_network_security_group.test.id
+}
+
+resource "azurerm_virtual_wan" "test" {
+  name                = "acctestvwan-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+}
+
+resource "azurerm_virtual_hub" "test" {
+  name                = "acctestvhub-%d"
```

Comment:
Could we use capitals in all resource names that allow it
```suggestion
  name                = "acctest-VHUB-%d"
```

Reply:
Updated